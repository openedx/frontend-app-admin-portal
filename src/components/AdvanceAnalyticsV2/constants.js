// Default group for the analytics. For now, it will be an empty string.
export const DEFAULT_GROUP = '';
