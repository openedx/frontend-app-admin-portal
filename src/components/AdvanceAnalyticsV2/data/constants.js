export const advancedAnalyticsV2QueryKeys = {
  chartCSV: (enterpriseId, chartName) => [...advancedAnalyticsV2QueryKeys.all, 'chartCSV', enterpriseId, chartName],
};

export const CHART_TYPES = {
  // Charts types for Skills tab
  BUBBLE: 'bubble',
  TOP_SKILLS_ENROLLMENT: 'top_skills_enrollment',
  TOP_SKILLS_COMPLETION: 'top_skills_completion',

  // Charts types for Completions tab
  COMPLETIONS_OVER_TIME: 'completions_over_time',
  TOP_COURSES_BY_COMPLETIONS: 'top_courses_by_completions',
  TOP_SUBJECTS_BY_COMPLETIONS: 'top_subjects_by_completions',

  // Charts types for Enrollments tab
  ENROLLMENTS_OVER_TIME: 'enrollments_over_time',
  TOP_COURSES_BY_ENROLLMENTS: 'top_courses_by_enrollments',
  TOP_SUBJECTS_BY_ENROLLMENTS: 'top_subjects_by_enrollments',
  INDIVIDUAL_ENROLLMENTS: 'individual_enrollments',

  // Charts types for Engagements tab
  ENGAGEMENTS_OVER_TIME: 'engagements_over_time',
  TOP_COURSES_BY_ENGAGEMENTS: 'top_courses_by_engagements',
  TOP_SUBJECTS_BY_ENGAGEMENTS: 'top_subjects_by_engagements',
  INDIVIDUAL_ENGAGEMENTS: 'individual_engagements',
};

export const ANALYTICS_TABS = {
  SKILLS: 'skills',
  COMPLETIONS: 'completions',
  ENROLLMENTS: 'enrollments',
  LEADERBOARD: 'leaderboard',
  ENGAGEMENTS: 'engagements',
};
