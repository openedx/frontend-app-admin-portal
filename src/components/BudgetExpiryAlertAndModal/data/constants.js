export const SEEN_ENTERPRISE_EXPIRATION_ALERT_COOKIE_PREFIX = 'seen-enterprise-expiration-alert-';

export const SEEN_ENTERPRISE_EXPIRATION_MODAL_COOKIE_PREFIX = 'seen-enterprise-expiration-modal-';

export const PLAN_EXPIRY_VARIANTS = {
  expired: 'Expired',
  expiring: 'Expiring',
};
