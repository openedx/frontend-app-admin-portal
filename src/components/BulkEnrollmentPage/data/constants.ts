export const SET_SELECTED_ROWS = 'SET SELECTED ROWS';
export const DELETE_ROW = 'DELETE ROW';
export const CLEAR_SELECTION = 'CLEAR SELECTION';
