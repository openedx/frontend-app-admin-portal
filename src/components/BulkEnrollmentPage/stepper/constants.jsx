export const NUM_CONTENT_ITEMS_PER_PAGE = 25;
export const STEPPER_TITLE = 'Learner enrollment';
export const ADD_LEARNERS_TITLE = 'Add learners';
export const REVIEW_TITLE = 'Review selections';
export const NEXT_BUTTON_TEXT = 'Next';
export const PREVIOUS_BUTTON_TEXT = 'Previous';
export const FINAL_BUTTON_TEXT = 'Enroll learners';
export const NEXT_BUTTON_TEST_ID = 'stepper-next';
export const PREV_BUTTON_TEST_ID = 'stepper-prev';
export const FINAL_BUTTON_TEST_ID = 'stepper-final';
export const NOTIFY_CHECKBOX_TEST_ID = 'notify-checkbox';
export const CUSTOMER_SUPPORT_HYPERLINK_TEST_ID = 'customer-support-hyperlink';
export const REVIEW_STEP = 'review';
export const ADD_COURSES_STEP = 'chooseCourses';
export const ADD_COURSES_TITLE = 'Add courses';
export const ALERT_MODAL_TITLE_TEXT = 'An error occurred behind the scenes';
export const ALERT_MODAL_BODY_TEXT = 'Some or all of your enrollments were not successful. Please '
                                     + 'wait a few minutes and try again. If the problem persists, '
                                     + 'please ';
export const SUPPORT_HYPERLINK_TEXT = 'contact enterprise customer support.';
export const SUPPORT_EMAIL_SUBJECT = 'An error occurred during Subscription Enrollment for enterprise ';
export const SUPPORT_EMAIL_BODY = 'The following error occurred when attempting to enroll learners: ';
export const ADD_COURSE_DESCRIPTION = 'By enrolling your learners in courses, you connect your'
                                      + ' learning community with the content that matters most '
                                      + 'to them, and take the guesswork out of getting started '
                                      + 'on the edX platform. To begin, select a course from '
                                      + 'your subscription catalog.';
export const ADD_LEARNERS_DESCRIPTION = 'Select learners with an active or pending subscription '
                                        + 'license to enroll. If you wish to enroll additional '
                                        + 'learners not shown, please first invite them under ';
export const WARNING_ALERT_TITLE_TEXT = 'Too many courses selected';
export const WARNING_ALERT_BODY_TEXT = 'We noticed that you are trying to enroll learners in over 7 '
                                        + 'courses at once - which might be too many for your learners '
                                        + 'to manage. If this is intentional, please continue with this '
                                        + 'enrollment process. If not, please go back and modify your '
                                        + 'course selections.';
