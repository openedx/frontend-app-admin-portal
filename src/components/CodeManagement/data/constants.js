// Tabs
export const MANAGE_CODES_TAB = 'manage-codes';
export const MANAGE_REQUESTS_TAB = 'manage-requests';

const MANAGE_CODES_TAB_LABEL = 'Manage Codes';
const MANAGE_REQUESTS_TAB_LABEL = 'Manage Requests';

/**
 * Used as tab values and in router params
 */
export const COUPON_CODE_TABS_VALUES = {
  [MANAGE_CODES_TAB]: MANAGE_CODES_TAB,
  [MANAGE_REQUESTS_TAB]: MANAGE_REQUESTS_TAB,
};

/**
 * Human readable tabs used on tab titles and browser helmet
 */
export const COUPON_CODE_TABS_LABELS = {
  [MANAGE_CODES_TAB]: MANAGE_CODES_TAB_LABEL,
  [MANAGE_REQUESTS_TAB]: MANAGE_REQUESTS_TAB_LABEL,
};

/** Default tab when no parameter is given */
export const DEFAULT_TAB = MANAGE_CODES_TAB;

/**
 * Url parameter that the set in the router
 * Consumed by useCurrentSubscriptionsTab hook to get tab value
 */
export const COUPON_CODE_TAB_PARAM = 'couponCodesTab';

/**
 * Generates coupon codes url matching from COUPON_CODE_TABS_VALUES
 * @example :couponCodesTab(tab0|tab1)?/
 */
const generatePathMatch = () => {
  const matchTabs = Object.values(COUPON_CODE_TABS_VALUES).join('|');
  return `:${COUPON_CODE_TAB_PARAM}(${matchTabs})?/`;
};

export const COUPON_CODES_PARAM_MATCH = generatePathMatch();
