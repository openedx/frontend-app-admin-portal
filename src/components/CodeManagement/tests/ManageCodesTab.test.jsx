import React from 'react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';

import ManageCodesTab from '../ManageCodesTab';

import { COUPONS_REQUEST, CLEAR_COUPONS } from '../../../data/constants/coupons';

const mockStore = configureMockStore([thunk]);
const initialState = {
  portalConfiguration: {
    enterpriseId: 'test-enterprise-id',
  },
  coupons: {
    loading: false,
    error: null,
    data: {
      count: 0,
      results: [],
    },
  },
  table: {
    'coupon-details': {},
  },
  form: {
    'code-assignment-modal-form': {
      values: {
        'email-address': '',
      },
    },
  },
};

const ManageCodesTabWrapper = props => (
  <MemoryRouter>
    <Provider store={props.store}>
      <ManageCodesTab
        location={{}}
        match={{
          path: '/test-page',
        }}
        history={{
          replace: () => {},
        }}
        {...props}
      />
    </Provider>
  </MemoryRouter>
);

ManageCodesTabWrapper.defaultProps = {
  store: mockStore({ ...initialState }),
};

ManageCodesTabWrapper.propTypes = {
  store: PropTypes.shape({}),
};

const sampleCouponData = {
  id: 0,
  title: 'test-title-1',
  start_date: '2019-01-03T23:23:51.581Z',
  end_date: '2019-09-18T14:46:36.716Z',
  errors: [],
  max_uses: 10,
  num_unassigned: 2,
  num_uses: 2,
  available: true,
  usage_limitation: 'Multi-use',
};

describe('ManageCodesTabWrapper', () => {
  describe('renders', () => {
    it('renders empty results correctly', () => {
      const tree = renderer
        .create((
          <ManageCodesTabWrapper />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders non-empty results correctly', () => {
      const store = mockStore({
        ...initialState,
        coupons: {
          ...initialState.coupons,
          data: {
            count: 2,
            num_pages: 1,
            results: [
              sampleCouponData,
              {
                ...sampleCouponData,
                id: 1,
                title: 'test-title-2',
              },
            ],
          },
        },
      });

      const tree = renderer
        .create((
          <ManageCodesTabWrapper store={store} />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders loading state correctly', () => {
      const store = mockStore({
        ...initialState,
        coupons: {
          ...initialState.coupons,
          loading: true,
        },
      });

      const tree = renderer
        .create((
          <ManageCodesTabWrapper store={store} />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders error state correctly', () => {
      const store = mockStore({
        ...initialState,
        coupons: {
          ...initialState.coupons,
          error: new Error('test error'),
        },
      });

      const tree = renderer
        .create((
          <ManageCodesTabWrapper store={store} />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('handles location.state on componentDidMount', () => {
    const wrapper = mount((
      <ManageCodesTabWrapper
        location={{
          state: {
            hasRequestedCodes: true,
          },
        }}
      />
    ));

    expect(wrapper.find('ManageCodesTab').instance().state.hasRequestedCodes).toBeTruthy();
  });

  it('handles overview_page query parameter change', () => {
    const store = mockStore({
      ...initialState,
      coupons: {
        ...initialState.coupons,
        data: {
          count: 100,
          num_pages: 2,
          results: [...Array(50)].map((_, index) => ({ ...sampleCouponData, id: index })),
        },
      },
    });
    const spy = jest.spyOn(store, 'dispatch');

    const wrapper = mount(<ManageCodesTabWrapper store={store} />);

    spy.mockRestore();

    wrapper.setProps({
      location: {
        search: '?overview_page=2',
      },
    });

    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('calls clearCouponOrders() on componentWillUnmount', () => {
    const store = mockStore({ ...initialState });

    const wrapper = mount(<ManageCodesTabWrapper store={store} />);
    wrapper.unmount();

    const actions = store.getActions();
    expect(actions.filter(action => action.type === CLEAR_COUPONS)).toHaveLength(1);
  });

  it('calls expand/collapse callbacks properly', () => {
    const store = mockStore({
      ...initialState,
      coupons: {
        ...initialState.coupons,
        data: {
          count: 1,
          num_pages: 1,
          results: [
            sampleCouponData,
            {
              ...sampleCouponData,
              id: 1,
              title: 'test-title-2',
            },
          ],
        },
      },
      table: {
        'coupon-details': {},
      },
      csv: {
        'coupon-details': {},
      },
    });

    const wrapper = mount(<ManageCodesTabWrapper store={store} />);
    const instance = wrapper.find('ManageCodesTab').instance();
    const spyExpand = jest.spyOn(instance, 'handleCouponExpand');
    const spyCollapse = jest.spyOn(instance, 'handleCouponCollapse');

    // expand
    wrapper.find('Coupon').first().find('.metadata').simulate('click');
    expect(spyExpand).toBeCalledTimes(1);

    // collapse
    wrapper.find('Coupon').first().find('.metadata').simulate('click');
    expect(spyCollapse).toBeCalledTimes(1);
  });

  it('fetches coupons on refresh button click', () => {
    const store = mockStore({ ...initialState });
    const wrapper = mount(<ManageCodesTabWrapper store={store} />);
    store.clearActions();
    wrapper.find('.fa-refresh').hostNodes().simulate('click');
    expect(store.getActions().filter(action => action.type === COUPONS_REQUEST)).toHaveLength(1);
  });
});
