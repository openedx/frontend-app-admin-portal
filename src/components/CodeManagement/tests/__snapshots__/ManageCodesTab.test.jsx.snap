// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageCodesTabWrapper renders renders empty results correctly 1`] = `
Array [
  <div
    className="row mt-4 mb-3 no-gutters"
  >
    <div
      className="col-12 col-xl-3 mb-3 mb-xl-0"
    >
      <h2>
        Overview
      </h2>
    </div>
    <div
      className="col-12 col-xl-4 mb-3 mb-xl-0"
    >
      <div
        className="pgn__searchfield d-flex"
      >
        <form
          className="d-flex align-items-center w-100"
          onReset={[Function]}
          onSubmit={[Function]}
          role="search"
        >
          <label
            className="m-0"
            htmlFor="pgn-searchfield-input-1"
          >
            <span
              className="sr-only"
            >
              search
            </span>
          </label>
          <input
            className="form-control"
            data-hj-suppress={true}
            disabled={false}
            id="pgn-searchfield-input-1"
            name="searchfield-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search by email or code..."
            role="searchbox"
            type="text"
            value=""
          />
          <button
            className="btn"
            disabled={false}
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-search "
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <span
              className="sr-only"
            >
              submit search
            </span>
          </button>
        </form>
      </div>
    </div>
    <div
      className="col-12 col-xl-5 mb-3 mb-xl-0 text-xl-right"
    >
      <button
        className="mr-2 btn btn-link"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden={true}
          className="fa fa-refresh mr-2"
          id="Icon2"
        />
        Refresh data
      </button>
      <a
        className="request-codes-btn btn btn-primary"
        href="/test-page/request-codes"
        onClick={[Function]}
      >
        <span
          aria-hidden={true}
          className="fa fa-plus mr-2"
          id="Icon3"
        />
        Request more codes
      </a>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      />
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="fade alert-content alert alert-warning show"
        role="alert"
      >
        <span
          className="pgn__icon alert-icon"
        >
          <svg
            aria-hidden={true}
            fill="none"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2L1 21h22L12 2z"
              fill="#F0CC00"
            />
            <path
              d="M13 16h-2v2h2v-2zM13 10h-2v4h2v-4z"
              fill="#111"
            />
          </svg>
        </span>
        <div
          className="pgn__alert-message-wrapper"
        >
          <div
            className="alert-message-content"
          >
            There are no results.
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`ManageCodesTabWrapper renders renders error state correctly 1`] = `
Array [
  <div
    className="row mt-4 mb-3 no-gutters"
  >
    <div
      className="col-12 col-xl-3 mb-3 mb-xl-0"
    >
      <h2>
        Overview
      </h2>
    </div>
    <div
      className="col-12 col-xl-4 mb-3 mb-xl-0"
    >
      <div
        className="pgn__searchfield d-flex"
      >
        <form
          className="d-flex align-items-center w-100"
          onReset={[Function]}
          onSubmit={[Function]}
          role="search"
        >
          <label
            className="m-0"
            htmlFor="pgn-searchfield-input-12"
          >
            <span
              className="sr-only"
            >
              search
            </span>
          </label>
          <input
            className="form-control"
            data-hj-suppress={true}
            disabled={false}
            id="pgn-searchfield-input-12"
            name="searchfield-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search by email or code..."
            role="searchbox"
            type="text"
            value=""
          />
          <button
            className="btn"
            disabled={false}
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-search "
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <span
              className="sr-only"
            >
              submit search
            </span>
          </button>
        </form>
      </div>
    </div>
    <div
      className="col-12 col-xl-5 mb-3 mb-xl-0 text-xl-right"
    >
      <button
        className="mr-2 btn btn-link"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden={true}
          className="fa fa-refresh mr-2"
          id="Icon13"
        />
        Refresh data
      </button>
      <a
        className="request-codes-btn btn btn-primary"
        href="/test-page/request-codes"
        onClick={[Function]}
      >
        <span
          aria-hidden={true}
          className="fa fa-plus mr-2"
          id="Icon14"
        />
        Request more codes
      </a>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      />
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="fade alert-content alert alert-danger show"
        role="alert"
      >
        <span
          className="pgn__icon alert-icon"
        >
          <svg
            aria-hidden={true}
            fill="none"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          className="pgn__alert-message-wrapper"
        >
          <div
            className="alert-message-content"
          >
            <div
              className="alert-heading h4"
            >
              Unable to load coupons
            </div>
            <p>
              Try refreshing your screen (
              test error
              )
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`ManageCodesTabWrapper renders renders loading state correctly 1`] = `
Array [
  <div
    className="row mt-4 mb-3 no-gutters"
  >
    <div
      className="col-12 col-xl-3 mb-3 mb-xl-0"
    >
      <h2>
        Overview
      </h2>
    </div>
    <div
      className="col-12 col-xl-4 mb-3 mb-xl-0"
    >
      <div
        className="pgn__searchfield d-flex"
      >
        <form
          className="d-flex align-items-center w-100"
          onReset={[Function]}
          onSubmit={[Function]}
          role="search"
        >
          <label
            className="m-0"
            htmlFor="pgn-searchfield-input-9"
          >
            <span
              className="sr-only"
            >
              search
            </span>
          </label>
          <input
            className="form-control"
            data-hj-suppress={true}
            disabled={false}
            id="pgn-searchfield-input-9"
            name="searchfield-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search by email or code..."
            role="searchbox"
            type="text"
            value=""
          />
          <button
            className="btn"
            disabled={false}
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-search "
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <span
              className="sr-only"
            >
              submit search
            </span>
          </button>
        </form>
      </div>
    </div>
    <div
      className="col-12 col-xl-5 mb-3 mb-xl-0 text-xl-right"
    >
      <button
        className="mr-2 btn btn-link"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden={true}
          className="fa fa-refresh mr-2"
          id="Icon10"
        />
        Refresh data
      </button>
      <a
        className="request-codes-btn btn btn-primary"
        href="/test-page/request-codes"
        onClick={[Function]}
      >
        <span
          aria-hidden={true}
          className="fa fa-plus mr-2"
          id="Icon11"
        />
        Request more codes
      </a>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      />
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="loading d-flex align-items-center justify-content-center coupons mt-3"
      >
        Loading...
        <span
          className="sr-only"
        >
          Loading
        </span>
      </div>
    </div>
  </div>,
]
`;

exports[`ManageCodesTabWrapper renders renders non-empty results correctly 1`] = `
Array [
  <div
    className="row mt-4 mb-3 no-gutters"
  >
    <div
      className="col-12 col-xl-3 mb-3 mb-xl-0"
    >
      <h2>
        Overview
      </h2>
    </div>
    <div
      className="col-12 col-xl-4 mb-3 mb-xl-0"
    >
      <div
        className="pgn__searchfield d-flex"
      >
        <form
          className="d-flex align-items-center w-100"
          onReset={[Function]}
          onSubmit={[Function]}
          role="search"
        >
          <label
            className="m-0"
            htmlFor="pgn-searchfield-input-4"
          >
            <span
              className="sr-only"
            >
              search
            </span>
          </label>
          <input
            className="form-control"
            data-hj-suppress={true}
            disabled={false}
            id="pgn-searchfield-input-4"
            name="searchfield-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search by email or code..."
            role="searchbox"
            type="text"
            value=""
          />
          <button
            className="btn"
            disabled={false}
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-search "
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <span
              className="sr-only"
            >
              submit search
            </span>
          </button>
        </form>
      </div>
    </div>
    <div
      className="col-12 col-xl-5 mb-3 mb-xl-0 text-xl-right"
    >
      <button
        className="mr-2 btn btn-link"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden={true}
          className="fa fa-refresh mr-2"
          id="Icon5"
        />
        Refresh data
      </button>
      <a
        className="request-codes-btn btn btn-primary"
        href="/test-page/request-codes"
        onClick={[Function]}
      >
        <span
          aria-hidden={true}
          className="fa fa-plus mr-2"
          id="Icon6"
        />
        Request more codes
      </a>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      />
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="coupon mb-3 mb-lg-2 rounded border"
      >
        <div
          aria-controls="coupon-details-0"
          aria-expanded={false}
          className="metadata row no-gutters p-2 d-flex align-items-center rounded"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
          >
            <small
              className="text-muted"
            >
              Coupon Name
            </small>
            <div>
              test-title-1
            </div>
          </div>
          <div
            className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col"
              >
                <small
                  className="text-muted"
                >
                  Valid From
                </small>
                <div>
                  January 3, 2019
                </div>
              </div>
              <div
                className="col"
              >
                <small
                  className="text-muted"
                >
                  Valid To
                </small>
                <div>
                  September 18, 2019
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col"
              >
                <small
                  className="text-muted"
                >
                  Assignments Remaining
                </small>
                <div>
                  2
                </div>
              </div>
              <div
                className="col"
              >
                <small
                  className="text-muted"
                >
                  Enrollments Redeemed
                </small>
                <div>
                  2 of 10
                  <span
                    className="ml-1"
                  >
                    (20%)
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
          >
            <span
              aria-hidden={true}
              className="fa fa-chevron-down text-primary"
              id="Icon7"
            />
            <span
              className="sr-only"
            >
              Open coupon details
            </span>
          </div>
        </div>
        <div
          className="coupon-details row no-gutters px-2 my-3 d-none"
          id="coupon-details-0"
        >
          <div
            className="col"
          />
        </div>
      </div>
      <div
        className="coupon mb-3 mb-lg-2 rounded border"
      >
        <div
          aria-controls="coupon-details-1"
          aria-expanded={false}
          className="metadata row no-gutters p-2 d-flex align-items-center rounded"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
          >
            <small
              className="text-muted"
            >
              Coupon Name
            </small>
            <div>
              test-title-2
            </div>
          </div>
          <div
            className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col"
              >
                <small
                  className="text-muted"
                >
                  Valid From
                </small>
                <div>
                  January 3, 2019
                </div>
              </div>
              <div
                className="col"
              >
                <small
                  className="text-muted"
                >
                  Valid To
                </small>
                <div>
                  September 18, 2019
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col"
              >
                <small
                  className="text-muted"
                >
                  Assignments Remaining
                </small>
                <div>
                  2
                </div>
              </div>
              <div
                className="col"
              >
                <small
                  className="text-muted"
                >
                  Enrollments Redeemed
                </small>
                <div>
                  2 of 10
                  <span
                    className="ml-1"
                  >
                    (20%)
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
          >
            <span
              aria-hidden={true}
              className="fa fa-chevron-down text-primary"
              id="Icon8"
            />
            <span
              className="sr-only"
            >
              Open coupon details
            </span>
          </div>
        </div>
        <div
          className="coupon-details row no-gutters px-2 my-3 d-none"
          id="coupon-details-1"
        >
          <div
            className="col"
          />
        </div>
      </div>
      <div
        className="d-flex mt-4 justify-content-center"
      >
        <nav
          aria-label="coupons pagination"
          className="pagination-default"
        >
          <div
            aria-atomic={true}
            aria-live="polite"
            aria-relevant="text"
            className="sr-only"
          >
            Page 1, Current Page, of 1
          </div>
          <ul
            className="pagination"
          >
            <li
              className="page-item disabled"
            >
              <button
                aria-label="Previous"
                className="previous page-link btn btn-primary"
                disabled={true}
                onClick={[Function]}
                tabIndex="-1"
                type="button"
              >
                <div>
                  <span
                    className="pgn__icon"
                  >
                    <svg
                      aria-hidden={true}
                      fill="none"
                      focusable={false}
                      height={24}
                      role="img"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12l4.58-4.59z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                  Previous
                </div>
              </button>
            </li>
            <li
              className="page-item disabled"
            >
              <button
                aria-label="Next"
                className="next page-link btn btn-primary"
                disabled={true}
                onClick={[Function]}
                tabIndex="-1"
                type="button"
              >
                <div>
                  Next
                  <span
                    className="pgn__icon"
                  >
                    <svg
                      aria-hidden={true}
                      fill="none"
                      focusable={false}
                      height={24}
                      role="img"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6-6-6z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </div>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>,
]
`;
